packages/devtools_app/lib/src/inspector/layout_explorer/box/box.dart

```diff
@@ -0,0 +1,428 @@
+                    size: Size(displayWidths[2], widgetHeight),
+                    offset: Offset(
+                        displayWidths[0] + displayWidths[1], displayHeights[0]),
+                    realSize: Size(widths[2], safeParentSize.height),
+                    layoutProperties: properties,
+                    isFreeSpace: true,
+                  ),
+                  horizontal: true,
+                ),
+              // Bottom padding.
+              if (heights[2] != null)
+                PaddingVisualizerWidget(
+                  RenderProperties(
+                    axis: Axis.horizontal,
+                    size: Size(widgetWidth, displayHeights[2]),
+                    offset: Offset(displayWidths[0],
+                        displayHeights[0] + displayHeights[1]),
+                    realSize: Size(safeParentSize.width, heights[2]),
+                    layoutProperties: properties,
+                    isFreeSpace: true,
+                  ),
+                  horizontal: false,
+                ),
+              BoxChildVisualizer(
+                isSelected: true,
+                state: this,
+                layoutProperties: properties,
+                renderProperties: RenderProperties(
+                  axis: Axis.horizontal,
+                  size: Size(widgetWidth, widgetHeight),
+                  offset: Offset(displayWidths[0], displayHeights[0]),
+                  realSize: properties.size,
+                  layoutProperties: properties,
+                ),
+              ),
+            ],
+          ),
+        );
+      },
+    );
+  }
+
+  LayoutProperties get parentProperties {
+    final parentElement = properties?.node?.parentRenderElement;
+    if (parentElement == null) return null;
+    final parentProperties = computeLayoutProperties(parentElement);
+    if (parentProperties.size == null) return null;
+    return parentProperties;
+  }
+
+  Widget _buildLayout(BuildContext context, BoxConstraints constraints) {
+    final maxHeight = constraints.maxHeight;
+    final maxWidth = constraints.maxWidth;
+
+    Widget widget = _buildChild(context);
+    final parentProperties = this.parentProperties;
+    if (parentProperties != null) {
+      // Wrap with a widget visualizer for the parent if there is a valid parent.
+      widget = WidgetVisualizer(
+        // TODO(jacobr): this node's name can be misleading more often than
+        // in the flex case the widget doesn't have its own RenderObject.
+        // Consider showing the true ancestor for the summary tree that first
+        // has a different render object.
+        title: describeBoxName(parentProperties),
+        largeTitle: true,
+        layoutProperties: parentProperties,
+        isSelected: false,
+        child: VisualizeWidthAndHeightWithConstraints(
+          properties: parentProperties,
+          warnIfUnconstrained: false,
+          child: Padding(
+            padding: const EdgeInsets.all(8.0),
```

Comment:
use defaultSpacing const (or whichever one is 8.0 in our theme consts)

Reply:
@pq has there ever been discussion of a lint to warn when a raw value is used where a const with the same value could be used?
It would be tricky to get right without false positives. For example you would probably need to see that EdgeInsets.all(denseSpacing) existed in the rest of the code base. This is probably more one of those lints that should be a code suggestion as I write code. As soon as I wrote 8 it should suggest the fix to the similar value used to define EdgeInsets.

Reply:
As it happens: https://github.com/dart-lang/linter/pull/2429 👍